cloup/_option_groups.py

```diff
@@ -185,19 +185,25 @@ def get_ungrouped_options(self, ctx: click.Context) -> Sequence[click.Option]:
 
     def get_argument_help_record(
         self, arg: click.Argument, ctx: click.Context
-    ) -> Tuple[str, str]:
+    ) -> Optional[Tuple[str, str]]:
         if isinstance(arg, cloup.Argument):
-            return arg.help_record(ctx)
+            return arg.get_help_record(ctx)
         return arg.make_metavar(), ""
 
     def get_arguments_help_section(self, ctx: click.Context) -> Optional[HelpSection]:
         if not any(getattr(arg, "help", None) for arg in self.arguments):
             return None
+        help_records = list(
+            filter(
+                None,
+                (self.get_argument_help_record(arg, ctx) for arg in self.arguments),
+            )
+        )
+        if not help_records:
+            return None
         return HelpSection(
             heading="Positional arguments",
-            definitions=[
-                self.get_argument_help_record(arg, ctx) for arg in self.arguments
-            ]
+            definitions=list(filter(None, help_records)),
```

Comment:
Filtering not needed. I'll refactor this method anyway.

Reply:
The idea is that `get_argument_help_record` can return `None` (given the signature of `get_help_record`), so it does in fact need to handle this case. Since a downstream user can subclass `Argument` or `Option` an override the `get_help_record` method.

Reply:
Sorry, I meant you already filtered at line 197.

Nonetheless... can you think of a reason why one would want to hide a positional argument from the help or worse from the usage string? I can't and I actually feel like that is something I don't want to allow. If you can't think of anything, I'd just "forbid" implementers from doing it by changing the return type to `Tuple[str, str]` (which is still fully compatible with the superclass signature).